geoips/commandline/geoips_command.py

```diff
@@ -71,13 +71,32 @@ def __init__(self, parent=None):
               Ex. GeoipsList would invoke this init function for each of its subcommand
               classes (GeoipsListPackages, GeoipsListScripts, ...). When it invokes this
               init, it supplies 'self' as an argument to follow the correct logic below.
+        legacy: optional - bool
+            - Whether or not the command supplied was called in legacy mode. This only
+              occurs for 'geoips run' commands, where instead of typing 'geoips run',
+              the user called 'run_procflow' or 'data_fusion_procflow'. This is used for
+              suppressing or displaying help information for '--procflow'.
         """
+        self.legacy = legacy
         self.github_org_url = "https://github.com/NRLMMD-GEOIPS/"
-        if parent:
+        self.parent = parent
+        if self.parent:
             # Set the combined name of the provided object. For example, if this was the
             # parent 'list' command, it would be 'geoips_list'. If it was a child of
             # list, for example 'scripts', combined name would be 'geoips_list_scripts'
             self.combined_name = f"{parent.combined_name}_{self.command_name}"
+
+            # We need to create a Geoips<cmd>Common Class for arguments
+            # that are shared between common commands. For example, we've created
+            # a 'GeoipsListCommon' class which adds arguments that will be shared
+            # by each GeoipsList<sub-cmd> class. Ie. if GeoipsListCommon has
+            # arguments --package, --columns, etc., and all of those arguments
+            # would be inherited by each GeoipsList<sub-cmd>
+            if "list" in self.combined_name:
+                parent_parsers = [GeoipsListCommon().subcommand_parser]
+            else:
+                parent_parsers = []
+
```

Comment:
We should consider how we can use `parser.parse_known_args()` to achieve this. I'll create an issue before merging this PR.

Reply:
This might be a good idea to tackle this in issue #580

Reply:
Unless it already exists, can you create a new issue discussing this and reference #580 from it? They're separate but related issues I think. #580 probably depends on this issue.

In this issue, we want to develop a method of adding parser options at different scopes including global (i.e. on the top-level `geoips` command) and for any command, sub-command, sub-sub-command, etc. I think we can probably achieve this somehow using `ArgumentParser.parse_known_args()`.

Reply:
See #637.